
package it.unibo.test.services;

/**
 * Please modify this class to meet your needs
 * This class is not complete
 */

import java.io.File;
import java.net.MalformedURLException;
import java.net.URL;
import javax.xml.namespace.QName;
import javax.jws.WebMethod;
import javax.jws.WebParam;
import javax.jws.WebResult;
import javax.jws.WebService;
import javax.xml.bind.annotation.XmlSeeAlso;
import javax.xml.ws.RequestWrapper;
import javax.xml.ws.ResponseWrapper;

/**
 * This class was generated by Apache CXF 2.5.2
 * 2018-10-08T22:52:10.368+02:00
 * Generated source version: 2.5.2
 * 
 */
public final class UniboServices_UniboServicesPort_Client {

    private static final QName SERVICE_NAME = new QName("http://services.unibo.it/", "UniboServicesService");

    private UniboServices_UniboServicesPort_Client() {
    }

    public static void main(String args[]) throws java.lang.Exception {
        URL wsdlURL = UniboServicesService.WSDL_LOCATION;
        if (args.length > 0 && args[0] != null && !"".equals(args[0])) { 
            File wsdlFile = new File(args[0]);
            try {
                if (wsdlFile.exists()) {
                    wsdlURL = wsdlFile.toURI().toURL();
                } else {
                    wsdlURL = new URL(args[0]);
                }
            } catch (MalformedURLException e) {
                e.printStackTrace();
            }
        }
      
        UniboServicesService ss = new UniboServicesService(wsdlURL, SERVICE_NAME);
        UniboServices port = ss.getUniboServicesPort();  
        
        {
        System.out.println("Invoking deletCredentialsFromIdUser...");
        int _deletCredentialsFromIdUser_idUser = 0;
        try {
            port.deletCredentialsFromIdUser(_deletCredentialsFromIdUser_idUser);

        } catch (Exception_Exception e) { 
            System.out.println("Expected exception: Exception has occurred.");
            System.out.println(e.toString());
        }
            }
        {
        System.out.println("Invoking getUser...");
        java.lang.Integer _getUser_idUser = null;
        try {
            it.unibo.test.services.UserBean _getUser__return = port.getUser(_getUser_idUser);
            System.out.println("getUser.result=" + _getUser__return);

        } catch (Exception_Exception e) { 
            System.out.println("Expected exception: Exception has occurred.");
            System.out.println(e.toString());
        }
            }
        {
        System.out.println("Invoking addUser...");
        it.unibo.test.services.UserBean _addUser_idUser = null;
        try {
            port.addUser(_addUser_idUser);

        } catch (Exception_Exception e) { 
            System.out.println("Expected exception: Exception has occurred.");
            System.out.println(e.toString());
        }
            }
        {
        System.out.println("Invoking addCredentials...");
        java.lang.Integer _addCredentials_idUser = null;
        it.unibo.test.services.CredentialBean _addCredentials_cred = null;
        try {
            port.addCredentials(_addCredentials_idUser, _addCredentials_cred);

        } catch (Exception_Exception e) { 
            System.out.println("Expected exception: Exception has occurred.");
            System.out.println(e.toString());
        }
            }
        {
        System.out.println("Invoking getCredential...");
        java.lang.Integer _getCredential_idUser = null;
        try {
            it.unibo.test.services.CredentialBean _getCredential__return = port.getCredential(_getCredential_idUser);
            System.out.println("getCredential.result=" + _getCredential__return);

        } catch (Exception_Exception e) { 
            System.out.println("Expected exception: Exception has occurred.");
            System.out.println(e.toString());
        }
            }
        {
        System.out.println("Invoking deleteUser...");
        int _deleteUser_idUser = 0;
        try {
            port.deleteUser(_deleteUser_idUser);

        } catch (Exception_Exception e) { 
            System.out.println("Expected exception: Exception has occurred.");
            System.out.println(e.toString());
        }
            }
        {
        System.out.println("Invoking deletCredentialsFromId...");
        int _deletCredentialsFromId_idUser = 0;
        try {
            port.deletCredentialsFromId(_deletCredentialsFromId_idUser);

        } catch (Exception_Exception e) { 
            System.out.println("Expected exception: Exception has occurred.");
            System.out.println(e.toString());
        }
            }

        System.exit(0);
    }

}
